### PR TITLE
Add quick-selection UI for sample SSP

### PR DESCRIPTION
**Pull request by danielnaab**
Implements #88 - To quickly load the sample SSP without using a file picker, this adds an unstyled button that links to the XML document on Github. This will always point to the document in the same branch that Federalist built the site from - support for which came from some light refactoring of how internal Github links are handled. Additionally - to improve stability as the frontend logic gets more complex - this branch includes the introduction of Overmind statemachine usage for the file selection/processing/reporting state transitions.

### CODE REVIEW

File: resources/validations/ui/src/context/browser/views/components/SSPValidator.tsx [hunk 1]
```diff
@@ -16,16 +17,34 @@ export const SSPValidator = () => {
         <span className="usa-hint" id="file-input-specific-hint">
           Select XML file
```
Comment (ohsh6o) [inline resources/validations/ui/src/context/browser/views/components/SSPValidator.tsx:18]: I would recommend we put something in the UI that is a little more contextually relevant for people not read into the purpose of the application.

I know this part of the UI and UX here, and it seems the `Drag file here` is from the USWDS, so I am not sure we want to customize that. I assume the headache is not pleasant but can we move the content underneath for the default file.

I am not sure this looks more busy, but I would love more feedback from the team. I just played with Firefox to edit the DOM in place after much trial and error.

Before:
Screen Shot 2021-06-24 at 11 47 30 AM


After:
Screen Shot 2021-06-24 at 11 46 22 AM

Thoughts @GaryGapinski and @rrkane10x?

**Review by ohsh6o — COMMENTED** [review]
Not sure my only constructive feedback is on the mark or off the mark. I also want some team feedback on that.

**Review by ohsh6o — APPROVED** [review]
LGTM.

File: resources/validations/ui/src/context/browser/views/components/SSPValidator.tsx (hunk 1 above)
Comment (GaryGapinski) [reply resources/validations/ui/src/context/browser/views/components/SSPValidator.tsx]: I tried this with the supplied document, one that I supplied with a specific schema association and one without an explicit schema association. All produced good output (and the schema association was properly ignored).

I noticed the results screen shows a few superfluous roles (warn, warning, info, information, unspecified). I have made these consistent (information, warning, error, and fatal) in #108.

**Review by GaryGapinski — APPROVED** [review]
_nihil obstat_

**Pull request merged**